packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -1635,9 +1635,14 @@ export function rootPathTo(construct: IConstruct, ancestor?: IConstruct): IConst
  * has only one component. Otherwise we fall back to the regular "makeUniqueId"
  * behavior.
  */
-function makeStackName(components: string[]) {
-  if (components.length === 1) { return components[0]; }
-  return makeUniqueResourceName(components, { maxLength: 128 });
+function makeStackName(components: string[], prefix: string='') {
+  if (components.length === 1) {
+    const stack_name = prefix + components[0];
+    if (stack_name.length <= 128) {
+      return prefix + components[0];
```

Comment:
Where there is a single component and a prefix, passing both the components and the prefix to the `makeUniqueResourceName` might seem like the simplest and most straight forward implementation, but there is an issue with this approach.
Doing so will make some tests fail, where a stage with an empty stack is defined. Doing so will generate a prefix (the stage name), and a single component with the name of the default stack. The `makeUniqueResourceName` removes all components with the `default` name, leaving us with no resources in the list, which makes `makeUniqueResourceName` throw an error.